### PR TITLE
dynamically load newest LTS version

### PR DESCRIPTION
**Pull request by fritzmg**
See #111

### CODE REVIEW

**Comment by fritzmg** [comment]
> I would prefer to call `$this->getCurrentLtsVersion()` only after the user has clicked "install" and not generally every time the user opens the Contao check.

It's not executed every time a user opens the Contao check. The LTS version is only fetched from contao.org when all of the following is true:

* user is on the install page
* `$this->hasInstallation()`
* `$this->canInstall()`
* `$this->canConnect()`
* `$this->canUsePhp()`

Only when the user is on the install page and the install controller determined, that the Contao version can be installed is the LTS version retrieved from contao.org.

> Then protected `$lts_version = '';` can be removed again I think.

Without the variable, I would have to fetch the LTS version more often from contao.org. I need it at least twice on the same page - once for the `<select>` value and once for the `<select>` text. However, I could change
```php
<?php if ($this->lts_version): ?>
```
in the template to
```php
<?php if (($lts_version = $this->getCurrentLtsVersion()): ?>
```
This way the LTS version would not be retrieved anymore, once the installer actually starts the installation (currently it would).

**Comment by leofeyer** [comment]
> It's not executed every time a user opens the Contao check. The LTS version is only fetched from contao.org when all of the following is true:
> 
> * user is on the install page
> * `$this->hasInstallation()`
> * `$this->canInstall()`
> * `$this->canConnect()`
> * `$this->canUsePhp()`
> 
> Only when the user is on the install page and the install controller determined, that the Contao version can be installed is the LTS version retrieved from contao.org.

Yes, but at this point the user might as well just check what the page is all about. Only after they have clicked the "install" button, we definitely know that they want to install and that we need to retrieve the version number.

It should not matter whether the user learns that they cannot connect before or after they have clicked the "install" button. But it will matter to the server whether the tool connects before or after the "install" button has been clicked.

> Then protected `$lts_version = '';` can be removed again I think.

This is only true if the version number is retrieved after the "install" button has been clicked.

**Comment by fritzmg** [comment]
> Yes, but at this point the user might as well just check what the page is all about. Only after they have clicked the "install" button, we definitely know that they want to install and that we need to retrieve the version number.

Ok, but this means the user will not actually know which version will be installed, when he clicks the install button. Only after the installation is complete, the check will actually tell him which Contao version has been installed. Should I implement it this way?

**Comment by leofeyer** [comment]
Hm, I see what you mean. On the other hand, shouldn't "latest LTS version" suffice?

**Comment by fritzmg** [comment]
> Hm, I see what you mean. On the other hand, shouldn't "latest LTS version" suffice?

It would suffice. Just seems less user friendly to me. Your decision... ;)

File: check/controller/installer.php [hunk 1]
```diff
@@ -36,6 +36,11 @@ class Installer
 	protected $message = '';
 
 	/**
+	 * @var string
+	 */
+	protected $lts_version = '';
```
Comment (leofeyer) [inline check/controller/installer.php]: This should be `$ltsVersion` according to the coding style.

**Comment by leofeyer** [comment]
Ok, the just leave it the way it is now. 😄

**Comment by fritzmg** [comment]
> This is only true if the version number is retrieved after the "install" button has been clicked.

When the install button is clicked, the version number is sent via POST (and is also referenced by that POST variable afterwards).

**Comment by leofeyer** [comment]
Yeah, just leave this stuff as is. 😄

**Pull request merged**